lib/literal.rb

```diff
@@ -25,15 +25,15 @@ def self.Enum(type)
 		end
 	end
 
-	def self.check(actual, expected, &context)
-		if expected === actual
+	def self.check(actual, expected, &)
+		raise ArgumentError.new("Cannot check type without a block") unless block_given?
+
+		if expected.respond_to?(:check)
+			expected.check(actual, &)
+		elsif expected === actual
```

Comment:
Should we run `===` first and only do `check` on failure? All failures will take twice as long. 🤔